src/Controls/src/Core/Shell/Shell.cs

```diff
@@ -1661,6 +1661,14 @@ static void OnCurrentItemChanged(BindableObject bindable, object oldValue, objec
 			shell.ShellController.AppearanceChanged(shell, false);
 			shell.ShellController.UpdateCurrentState(ShellNavigationSource.ShellItemChanged);
 
+			// If the new ShellItem is a FlyoutItem or a ShellGroupItem, we need to update the QueryAttributes
+			var existing = (ShellRouteParameters)shell.CurrentContent?.GetValue(ShellContent.QueryAttributesProperty);
+            if(oldValue is not null && newValue is not null && (shell.CurrentItem is FlyoutItem || shell.CurrentItem is ShellGroupItem) && existing is null)
```

Comment:
do we need to check for oldValue ?

Reply:
Yes @StephaneDelcroix , The `ApplyQueryAttributes trigger` is not required for the initial load; therefore, a null check has been added for the `oldValue`.